scripts/build-controller-image.sh

```diff
@@ -17,43 +16,27 @@ check_is_installed docker
 
 USAGE="
 Usage:
```

Comment:
One more thing we can take care of in this PR, is changing how this script is called in other files. PTAL at [./kind-build-test.sh](https://github.com/aws/aws-controllers-k8s/blob/main/scripts/kind-build-test.sh#L140-L145) and [./publish-controller-image.sh](https://github.com/aws/aws-controllers-k8s/blob/main/scripts/publish-controller-image.sh#L57-L60)

Reply:
@A-Hilaly  public-controller-image.sh calls this script setting the svc docker image tag to AWS_SERVICE_DOCKER_IMG_TAG, this script expects var name AWS_SERVICE_DOCKER_IMG instead, there seems to be some confusion and the script (public-controller-image.sh) uses both variable names indistinctly, is that correct ?

Reply:
Nice catch. This is correct - `AWS_SERVICE_DOCKER_IMG` is the variable that should be used to call `./build-controller-image.sh`. the other variable `AWS_SERVICE_DOCKER_IMG_TAG` should only be used to construct the variable `AWS_SERVICE_DOCKER_IMG`

Reply:
Something like this makes more sense
```bash
: "${AWS_SERVICE_DOCKER_IMG:="$DOCKER_REPOSITORY:$DEFAULT_AWS_SERVICE_DOCKER_IMG_TAG"}"

$SCRIPTS_DIR/build-controller-image.sh -s "$AWS_SERVICE" -i "$AWS_SERVICE_DOCKER_IMG"
```